rust/lance-datafusion/src/utils.rs

```diff
@@ -46,12 +46,21 @@ pub async fn reader_to_stream(
     .await
     .unwrap()?;
     schema.validate()?;
+    let reader = RecordBatchIterator::new(peekable, arrow_schema);
+    Ok((
+        Box::new(reader) as Box<dyn RecordBatchReader + Send>,
+        schema,
+    ))
+}
 
+/// Convert reader to a stream.
+///
+/// The reader will be called in a background thread.
+pub fn reader_to_stream(batches: Box<dyn RecordBatchReader + Send>) -> SendableRecordBatchStream {
```

Comment:
Found we sometimes didn't need to peek the reader, so separated this into two different functions: `peek_reader_schema()` and `reader_to_stream()`.